Samples~/LootLockerExamples/8 - Virtual Currency Purchases/LootLockerVirtualStore.cs

```diff
@@ -0,0 +1,126 @@
+using UnityEngine;
+using UnityEngine.UI;
+using LootLocker.Requests;
+
+namespace LootLocker
+{
+    public class LootLockerVirtualStore : MonoBehaviour
+    {
+        public static LootLockerVirtualStore Instance { get; private set; }
+
+        public Transform storeUI;
+        public Transform virtualPurchaseProductPrefab;
+
+        public Text goldAmount;
+
+        public string catalogKey = "";
+        public string walletID = "";
+
+        private string currencyCode;
+
+        public void Awake()
+        {
+
+            if (Instance == null)
+            {
+                Instance = this;
+            }
+            else Destroy(gameObject);
+
+            LootLockerSettingsOverrider.OverrideSettings();
+
+            LootLockerSDKManager.StartGuestSession((response) =>
+            {
+                if (!response.success)
+                {
+                    Debug.Log("Could not start guest session");
+                    return;
+                }
+                FetchPlayerWallet(response.player_ulid);
+                FetchVirtualProductsFromCatalog();
+
+            });
+        }
+
+        public void FetchPlayerWallet(string ulid)
+        {
+            LootLockerSDKManager.GetWalletByHolderId(ulid, LootLockerEnums.LootLockerWalletHolderTypes.player, (response) =>
+            {
+                if (!response.success)
+                {
+                    Debug.Log("Could not start guest session");
+                    return;
+                }
+
+                walletID = response.id;
+                GetPlayerGold();
+                GiveGold();
+            });
+        }
+
+        public void GetPlayerGold()
+        {
+            LootLockerSDKManager.ListBalancesInWallet(walletID, (response) =>
+            {
+                if (!response.success)
+                {
+                    Debug.Log("Could not list balances");
+                    return;
+                }
+
+                currencyCode = response.balances[0].currency.code.ToUpper() + ": ";
+
+                goldAmount.text = currencyCode + response.balances[0].amount;
+            });
+        }
+
+
+        public void FetchVirtualProductsFromCatalog()
+        {
+            LootLockerSDKManager.ListCatalogItems(catalogKey, 0, "", (response) =>
```

Comment:
From what I can tell you never set catalogKey anywhere??

(after having read the code a bit more)
Oh, nvm. You set it in the editor? Add a comment by the field that it should be set in editor and make walletId private. Because they are next to each other and look the same but are used differently.

Reply:
You're right thank you!